main.js

```diff
@@ -92,24 +92,36 @@ function getMainMapAddrs(json) {
     return addrs;
 }
 
-function drawMainMap(addrList) {
-    let map = L.map('main-map', {
-        center: [44.4781994, -73.2126357],
-        zoom: 20
-    });
+async function drawMainMap(addrList) {
+    let map = L.map('main-map').setView([44.4781994, -73.2126357],15);
 
     L.tileLayer('https://{s}.tile.openstreetmap.fr/osmfr/{z}/{x}/{y}.png', {
         maxZoom: 20,
         attribution: '&copy; Openstreetmap France | &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
     }).addTo(map);
 
-    addrList.forEach((ad) => {
-        const latArr = getCoords(ad.addr);
 
-        console.log(latArr);
-        let mark = L.marker([latArr.lat, latArr.lng]).addTo(map);
-        return mark;
-    })
+    for await (const ad of addrList) {
+        const latArr = await getCoords(ad.addr);
+
+        console.log({latArr});
+        let { lat, lon } = latArr;
+        console.log({lat})
+        console.log({lon})
+        latsLongs.push(latArr)
+        let mark = L.marker([lat, lon]).addTo(map);
+    }
+
+    // addrList.forEach((ad) => {
+    //     const latArr = await getCoords(ad.addr);
+
+    //     console.log({latArr});
+    //     let { lat, lon } = latArr;
+    //     console.log({lat})
+    //     console.log({lon})
+    //     let mark = L.marker([lat, lon]).addTo(map);
+    //     return mark;
+    // })
```

Comment:
This bit of code wont do what you expect because of the way fetching and promises work. Sort of like I explained in the message on slack, the ```forEach``` will loop through the ```addrList``` variable without waiting for any fetches to go get the data and resolve the promise (which is data that you can't work with). So it will loop through all of the addresses and call ```getCoords``` for as many addresses there are, almost all at once. Then when one of the fetches is complete, it will return it to ```latArr``` and then maybe place a marker. If you run that code, you will see that ```latArr``` is logged out only once, even though all of the fetches are made.